rules/rules-reviewed/hibernate/hibernate-search.windup.xml

```diff
@@ -0,0 +1,1489 @@
+                    <message>
+                        The equivalent to Hibernate Search 5’s SearchFactory is Hibernate Search 6’s SearchMapping, but some operations are more conveniently accessible directly from SearchSession.
+                        Check the link for more information and examples.
+                    </message>
+                    <link href="https://docs.jboss.org/hibernate/search/6.0/migration/html_single/#_searchfactory" title="Hibernate Search 6 Reference - SearchFactory"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hibernate-search-01020">
+            <when>
+                <javaclass references="org.hibernate.search.MassIndexer"/>
+            </when>
+            <perform>
+                <hint title="MassIndexer has moved" category-id="mandatory" effort="1">
+                    <message>
+                        The MassIndexer mostly stays the same in Hibernate Search 6, but it moved to a different package: `org.hibernate.search.mapper.orm.massindexing.MassIndexer`.
+                        There are some other changes that can be checked by following the link.
+                    </message>
+                    <link href="https://docs.jboss.org/hibernate/search/6.0/migration/html_single/#massindexer" title="Hibernate Search 6 Reference - MassIndexer"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hibernate-search-01030">
+            <when>
+                <javaclass references="org.hibernate.search.jsr352.massindexing.MassIndexingJob"/>
+            </when>
+            <perform>
+                <hint title="MassIndexer has moved" category-id="mandatory" effort="1">
+                    <message>
+                        The mass indexing Batch (JSR-352) job mostly stays the same in Hibernate Search 6, but `MassIndexingJob` moved to a different package: `org.hibernate.search.batch.jsr352.core.massindexing.MassIndexingJob`.
+                    </message>
+                    <link href="https://docs.jboss.org/hibernate/search/6.0/migration/html_single/#jsr352" title="Hibernate Search 6 Reference - MassIndexingJob"/>
+                </hint>
+            </perform>
+        </rule>
+
+        <!-- DEPENDENCIES -->
+        <rule id="hibernate-search-01040">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="org.hibernate" artifactId="hibernate-search{*}"/>
+                    </project>
+                    <dependency groupId="org.hibernate" artifactId="hibernate-search{*}"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Dependency must be migrated from Hibernate Search 5 to 6" effort="1" category-id="mandatory">
+                    <message>Hibernate Search 6 uses the groupId `org.hibernate.search` and all dependencies must be updated</message>
```

Comment:
You might want to add this link, since there are some quirks for Lucene users: https://docs.jboss.org/hibernate/search/6.0/migration/html_single/#maven-coordinates